src/objectc.c

```diff
@@ -125,6 +125,44 @@ int msgpack_pack_object(msgpack_packer* pk, msgpack_object d)
     }
 }
 
+bool msgpack_object_init(msgpack_object* d, int type, void* data, size_t size)
+{
+    d->type = type;
+    switch (type) {
+    case MSGPACK_OBJECT_STR:
+        {
+            d->via.str.ptr = (char *)data;
+            d->via.str.size = size;
+            break;
+        }
+    case MSGPACK_OBJECT_BIN:
+        {
+            d->via.bin.ptr = (char *)data;
+            d->via.bin.size = size;
+            break;
+        }
+    case MSGPACK_OBJECT_ARRAY:
+        {
+            d->via.array.ptr = (msgpack_object *)data;
+            d->via.array.size = size;
+            break;
+        }
+    case MSGPACK_OBJECT_MAP:
+        {
+            d->via.map.ptr = (msgpack_object_kv *)data;
+            d->via.map.size = size;
+            break;
+        }
+    default:
+        {
+            // Other types are not supported and need to be initialized manually.
+            d->type = MSGPACK_OBJECT_NIL;
```

Comment:
```suggestion
```

Sorry for pointing things out in a scattered way.
I noticed that if the function returns false, the destination object `d` shouldn't be updated.
Could you remove this line?

Reply:
I added this line just to return a proper nill object back. But if its misleading then i will remove it

Reply:
Like incase of failure my code would have assigned d->type = type at the start of function. Since we dont have a failure type i have assigned nil. is it fine if i assign -1 instead of nil type? @redboltz

Reply:
If the function fails (returns false), it's preferable if it doesn't cause any side effects, if possible.
So in this case, do nothing for `d`.

Reply:
The PR is for C code but this is similar concept as C++ exception guaranty (Strong exception guarantee).
See https://en.cppreference.com/w/cpp/language/exceptions (Exception Safety 2)